### ISSUE TITLE
Crash during table fuzzing - Guarantee failed: [!pair.first.inner.overlaps(region.inner)]

### ISSUE THREAD

**Issue by Tryneus** [rethinkdb/rethinkdb#4761]
Appears to have been during the creation of a new changefeed.  This guarantee failed on two of the cores of the crashing server.  Fuzzer parameters: 4 servers, 16 clients, changefeeds enabled.  I've been running the fuzzer quite a bit and this is the first time I've seen it, so this is probably a fairly small timing window.

```
error: Error in ../src/rdb_protocol/store.cc at line 894:
error: Guarantee failed: [!pair.first.inner.overlaps(region.inner)]
error: Backtrace:
error: Mon Aug 24 18:51:42 2015

1: rethinkdb_backtrace(void**, int) at rethinkdb_backtrace.cc:100
2: backtrace_t::backtrace_t() at backtrace.cc:203
3: lazy_backtrace_formatter_t::lazy_backtrace_formatter_t() at backtrace.cc:283
4: format_backtrace(bool) at backtrace.cc:198
5: report_fatal_error(char const*, int, char const*, ...) at errors.cc:83
6: store_t::get_or_make_changefeed_server(hash_region_t<key_range_t> const&) at store.cc:894
7: rdb_read_visitor_t::operator()(changefeed_subscribe_t const&) at store.cc:262
8: void boost::detail::variant::invoke_visitor<rdb_read_visitor_t>::internal_visit<changefeed_subscribe_t const>(changefeed_subscribe_t const&, int) at variant.hpp:858
9: boost::detail::variant::invoke_visitor<rdb_read_visitor_t>::result_type boost::detail::variant::visitation_impl_invoke_impl<boost::detail::variant::invoke_visitor<rdb_read_visitor_t>, void const*, changefeed_subscribe_t>(int, boost::detail::variant::invoke_visitor<rdb_read_visitor_t>&, void const*, changefeed_subscribe_t*, mpl_::bool_<false>) at visitation_impl.hpp:144
10: boost::detail::variant::invoke_visitor<rdb_read_visitor_t>::result_type boost::detail::variant::visitation_impl_invoke<boost::detail::variant::invoke_visitor<rdb_read_visitor_t>, void const*, changefeed_subscribe_t, boost::variant<point_read_t, rget_read_t, intersecting_geo_read_t, nearest_geo_read_t, changefeed_subscribe_t, changefeed_stamp_t, changefeed_limit_subscribe_t, changefeed_point_stamp_t, distribution_read_t, dummy_read_t, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_>::has_fallback_type_>(int, boost::detail::variant::invoke_visitor<rdb_read_visitor_t>&, void const*, changefeed_subscribe_t*, boost::variant<point_read_t, rget_read_t, intersecting_geo_read_t, nearest_geo_read_t, changefeed_subscribe_t, changefeed_stamp_t, changefeed_limit_subscribe_t, changefeed_point_stamp_t, distribution_read_t, dummy_read_t, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_>::has_fallback_type_, int) at visitation_impl.hpp:170
11: boost::detail::variant::invoke_visitor<rdb_read_visitor_t>::result_type boost::detail::variant::visitation_impl<mpl_::int_<0>, boost::detail::variant::visitation_impl_step<boost::mpl::l_iter<boost::mpl::l_item<mpl_::long_<10l>, point_read_t, boost::mpl::l_item<mpl_::long_<9l>, rget_read_t, boost::mpl::l_item<mpl_::long_<8l>, intersecting_geo_read_t, boost::mpl::l_item<mpl_::long_<7l>, nearest_geo_read_t, boost::mpl::l_item<mpl_::long_<6l>, changefeed_subscribe_t, boost::mpl::l_item<mpl_::long_<5l>, changefeed_stamp_t, boost::mpl::l_item<mpl_::long_<4l>, changefeed_limit_subscribe_t, boost::mpl::l_item<mpl_::long_<3l>, changefeed_point_stamp_t, boost::mpl::l_item<mpl_::long_<2l>, distribution_read_t, boost::mpl::l_item<mpl_::long_<1l>, dummy_read_t, boost::mpl::l_end> > > > > > > > > > >, boost::mpl::l_iter<boost::mpl::l_end> >, boost::detail::variant::invoke_visitor<rdb_read_visitor_t>, void const*, boost::variant<point_read_t, rget_read_t, intersecting_geo_read_t, nearest_geo_read_t, changefeed_subscribe_t, changefeed_stamp_t, changefeed_limit_subscribe_t, changefeed_point_stamp_t, distribution_read_t, dummy_read_t, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_>::has_fallback_type_>(int, int, boost::detail::variant::invoke_visitor<rdb_read_visitor_t>&, void const*, mpl_::bool_<false>, boost::variant<point_read_t, rget_read_t, intersecting_geo_read_t, nearest_geo_read_t, changefeed_subscribe_t, changefeed_stamp_t, changefeed_limit_subscribe_t, changefeed_point_stamp_t, distribution_read_t, dummy_read_t, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_>::has_fallback_type_, mpl_::int_<0>*, boost::detail::variant::visitation_impl_step<boost::mpl::l_iter<boost::mpl::l_item<mpl_::long_<10l>, point_read_t, boost::mpl::l_item<mpl_::long_<9l>, rget_read_t, boost::mpl::l_item<mpl_::long_<8l>, intersecting_geo_read_t, boost::mpl::l_item<mpl_::long_<7l>, nearest_geo_read_t, boost::mpl::l_item<mpl_::long_<6l>, changefeed_subscribe_t, boost::mpl::l_item<mpl_::long_<5l>, changefeed_stamp_t, boost::mpl::l_item<mpl_::long_<4l>, changefeed_limit_subscribe_t, boost::mpl::l_item<mpl_::long_<3l>, changefeed_point_stamp_t, boost::mpl::l_item<mpl_::long_<2l>, distribution_read_t, boost::mpl::l_item<mpl_::long_<1l>, dummy_read_t, boost::mpl::l_end> > > > > > > > > > >, boost::mpl::l_iter<boost::mpl::l_end> >*) at visitation_impl.hpp:256
12: boost::detail::variant::invoke_visitor<rdb_read_visitor_t>::result_type boost::variant<point_read_t, rget_read_t, intersecting_geo_read_t, nearest_geo_read_t, changefeed_subscribe_t, changefeed_stamp_t, changefeed_limit_subscribe_t, changefeed_point_stamp_t, distribution_read_t, dummy_read_t, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_>::internal_apply_visitor_impl<boost::detail::variant::invoke_visitor<rdb_read_visitor_t>, void const*>(int, int, boost::detail::variant::invoke_visitor<rdb_read_visitor_t>&, void const*) at variant.hpp:1797
13: boost::detail::variant::invoke_visitor<rdb_read_visitor_t>::result_type boost::variant<point_read_t, rget_read_t, intersecting_geo_read_t, nearest_geo_read_t, changefeed_subscribe_t, changefeed_stamp_t, changefeed_limit_subscribe_t, changefeed_point_stamp_t, distribution_read_t, dummy_read_t, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_>::internal_apply_visitor<boost::detail::variant::invoke_visitor<rdb_read_visitor_t> >(boost::detail::variant::invoke_visitor<rdb_read_visitor_t>&) const at variant.hpp:1823
14: rdb_read_visitor_t::result_type boost::variant<point_read_t, rget_read_t, intersecting_geo_read_t, nearest_geo_read_t, changefeed_subscribe_t, changefeed_stamp_t, changefeed_limit_subscribe_t, changefeed_point_stamp_t, distribution_read_t, dummy_read_t, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_>::apply_visitor<rdb_read_visitor_t>(rdb_read_visitor_t&) const at variant.hpp:1846
15: rdb_read_visitor_t::result_type boost::apply_visitor<rdb_read_visitor_t, boost::variant<point_read_t, rget_read_t, intersecting_geo_read_t, nearest_geo_read_t, changefeed_subscribe_t, changefeed_stamp_t, changefeed_limit_subscribe_t, changefeed_point_stamp_t, distribution_read_t, dummy_read_t, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_> const>(rdb_read_visitor_t&, boost::variant<point_read_t, rget_read_t, intersecting_geo_read_t, nearest_geo_read_t, changefeed_subscribe_t, changefeed_stamp_t, changefeed_limit_subscribe_t, changefeed_point_stamp_t, distribution_read_t, dummy_read_t, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_> const&) at apply_visitor_unary.hpp:60
16: store_t::protocol_read(read_t const&, read_response_t*, real_superblock_t*, signal_t*) at store.cc:583
17: store_t::read(metainfo_checker_t const&, read_t const&, read_response_t*, read_token_t*, signal_t*) at btree_store.cc:189
18: store_subview_t::read(metainfo_checker_t const&, read_t const&, read_response_t*, read_token_t*, signal_t*) at store_subview.hpp:104
19: replica_t::do_read(read_t const&, state_timestamp_t, signal_t*, read_response_t*) at replica.cc:43
20: local_replicator_t::do_read(read_t const&, state_timestamp_t, signal_t*, read_response_t*) at local_replicator.cc:84
21: primary_dispatcher_t::read(read_t const&, fifo_enforcer_sink_t::exit_read_t*, order_token_t, signal_t*, read_response_t*) at primary_dispatcher.cc:136
22: primary_execution_t::on_read(read_t const&, fifo_enforcer_sink_t::exit_read_t*, order_token_t, signal_t*, read_response_t*, admin_err_t*) at exec_primary.cc:427
23: non-virtual thunk to primary_execution_t::on_read(read_t const&, fifo_enforcer_sink_t::exit_read_t*, order_token_t, signal_t*, read_response_t*, admin_err_t*) at exec_primary.cc:443
...
```

**Comment by mlucy** [comment]
Er, I'm not exactly sure how this slipped off my plate for 4 months, but we've fixed a lot of changefeed bugs in the meantime.  @Tryneus -- is there an easy way to test whether this still happens?

**Comment by Tryneus** [comment]
@mlucy, I can spend some time running the fuzzer against `2.2.x` and see if anything shows up, but beyond that, these can be pretty hard to reproduce.

**Comment by danielmewes** [comment]
Looks like a duplicate of https://github.com/rethinkdb/rethinkdb/issues/5576 , which we have hopefully fixed in 2.3.0.
